src/ILCompiler.Build.Tasks/src/ComputeManagedAssemblies.cs

```diff
@@ -0,0 +1,177 @@
+        /// in the app's closure.
+        /// </summary>
+        [Required]
+        public ITaskItem[] FrameworkAssemblies
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The native apphost (whose name ends up colliding with the CoreRT output binary) 
+        /// </summary>
+        [Required]
+        public string DotNetAppHostExecutableName
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The CoreCLR dotnet host fixer library that can be skipped during publish
+        /// </summary>
+        [Required]
+        public string DotNetHostFxrLibraryName
+        {
+            get;
+            set;
+        }
+
+
+        /// <summary>
+        /// The CoreCLR dotnet host policy library that can be skipped during publish
+        /// </summary>
+        [Required]
+        public string DotNetHostPolicyLibraryName
+        {
+            get;
+            set;
+        }
+
+        [Output]
+        public ITaskItem[] ManagedAssemblies
+        {
+            get;
+            set;
+        }
+
+        [Output]
+        public ITaskItem[] AssembliesToSkipPublish
+        {
+            get;
+            set;
+        }
+
+        public override bool Execute()
+        {
+            var list = new List<ITaskItem>();
+            var assembliesToSkipPublish = new List<ITaskItem>();
+
+            ITaskItem[] assemblies = this.Assemblies;
+
+            var coreRTFrameworkAssembliesToUse = new HashSet<string>();
+
+            foreach (var x in SdkAssemblies)
+            {
+                coreRTFrameworkAssembliesToUse.Add(Path.GetFileName(x.ItemSpec));
+            }
+
+            foreach (var x in FrameworkAssemblies)
+            {
+                coreRTFrameworkAssembliesToUse.Add(Path.GetFileName(x.ItemSpec));
+            }
+
+            for (int i = 0; i < assemblies.Length; i++)
+            {
+                ITaskItem taskItem = assemblies[i];
+
+                // Skip crossgen images
+                if (taskItem.ItemSpec.EndsWith(".ni.dll", StringComparison.OrdinalIgnoreCase))
+                {
+                    assembliesToSkipPublish.Add(taskItem);
+                    continue;
+                }
+
+                // Skip the native apphost (whose name ends up colliding with the CoreRT output binary) and supporting libraries
+                if (taskItem.ItemSpec.EndsWith(DotNetAppHostExecutableName, StringComparison.OrdinalIgnoreCase) || taskItem.ItemSpec.Contains(DotNetHostFxrLibraryName) || taskItem.ItemSpec.Contains(DotNetHostPolicyLibraryName))
+                {
+                    assembliesToSkipPublish.Add(taskItem);
+                    continue;
+                }
+
+                // Prototype aid - remove the native CoreCLR runtime pieces from the publish folder
+                // TODO: Is this a correct approach? 
```

Comment:
@nattress Could I ask you to take a look at this?

Reply:
After testing out a couple of scenarios - it breaks after publishing (native dlls are filtered out). That was exactly what I was looking for, thanks.